tsconfig.esm.json

```diff
@@ -0,0 +1,16 @@
+{
+  "compilerOptions": {
```

Comment:
Let's use [config inheritance](https://www.typescriptlang.org/tsconfig#extends) here instead of copying all the settings

Reply:
I grouped them into a `configs/` - let me know if you'd prefer to organize these in a different way (an alternative is tsconfig.esm.json directly extends tsconfig.cjs.json)